src/nsrt_learning/side_predicate_learning.py

```diff
@@ -76,12 +89,16 @@ def _recompute_datastores_from_segments(
                         if not ground_op.preconditions.issubset(
                                 segment.init_atoms):
                             continue
-                        # Check if effects match. Note that we're using the side
-                        # predicates semantics here!
-                        atoms = utils.apply_operator(ground_op,
-                                                     segment.init_atoms)
-                        if not atoms.issubset(segment.final_atoms):
-                            continue
+                        # Check if effects match.
+                        if semantics == "apply_operator":
+                            atoms = utils.apply_operator(
+                                ground_op, segment.init_atoms)
+                            if not atoms.issubset(segment.final_atoms):
+                                continue
+                        elif semantics == "add_effects":
```

Comment:
We added these new semantics because we want to use them when recomputing the datastore of a PNAD after updating its add effects. I'm not convinced it's necessary or correct.

Reply:
we did this because we saw a case where it mattered, although i can't seem to replicate empirically (even buggy code can work fine on RNT so it's hard to tell). @NishanthJKumar do you remember